lib/mix/tasks/phx.gen.notifier.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Mix.Tasks.Phx.Gen.Notifier do
+  @shortdoc "Generates a notifier to send emails using the mailer"
+
+  @moduledoc """
+  Generates a notifier to send emails using the mailer
+
+      mix phx.gen.notifier Accounts User welcome_user reset_password confirmation_instructions
+
+  This task expects a context module name, followed by a
+  notifier name and one or more message names. Messages
+  are the functions that will be created prefixed by "deliver",
+  so the message name should be "snake_case" without punctuation.
+
+  Note that this task expects that you have a mailer defined
+  as "YourApp.Mailer". Check the `mix phx.new` task for more details.
+  """
+
+  use Mix.Task
+
+  @switches [
+    context: :boolean,
+    context_app: :string,
+    prefix: :string
+  ]
+
+  @default_opts [context: true]
+
+  alias Mix.Phoenix.Context
+
+  @doc false
+  def run(args) do
+    if Mix.Project.umbrella?() do
+      Mix.raise(
+        "mix phx.gen.notifier must be invoked from within your *_web application root directory"
```

Comment:
Shouldn't email templates be separated from web?

Reply:
They are but the generators always run from web because thats where phoenix is defined.